### ISSUE TITLE
printing to standard streams triggers context switch to a Task

### ISSUE THREAD

**Issue by carlobaldassi** [JuliaLang/julia#27168]
I have found this within a producer/consumer pattern. In summary, if the consumer uses `print` to either `stdout` or `stderr`, execution jumps back to the producer. (This does not happen when printing to a file.)

Here is a reduced example:

```julia
# crude logging function `fp`
tmpfile = tempname()
fp(args...) = open(tmpfile, "a") do f
    println(f, args...)
end
open(tmpfile, "w") do f; end


function producer(c::Channel)
    k = 2
    while k > 0
        fp("[P] 1 k=$k")
        put!(c, k)
        fp("[P] 2")
        k -= 1
        fp("[P] 3")
    end
end

function tst()
    for k in Channel(producer)
        fp("{C} 1 k=$k")
        fp("{C} 2")
        #println("A") # uncommenting this -> switch to `producer`
        fp("{C} 3")
    end
end

tst()
println("LOG:")
print(read(tmpfile, String))
rm(tmpfile)
```

Without the `print` in `tst`, everything goes how I'd expect, i.e. the producer is only invoked by the `for` loop and all context switches only take place between steps `{C}3` and `{C}1`:

```
LOG:
[P] 1 k=2
{C} 1 k=2
{C} 2
{C} 3
[P] 2
[P] 3
[P] 1 k=1
{C} 1 k=1
{C} 2
{C} 3
[P] 2
[P] 3
```

Instead, with the `print` call, the producer takes back control between steps `{C}2` and `{C}3`:

```
LOG:
[P] 1 k=2
{C} 1 k=2
{C} 2
[P] 2
[P] 3
[P] 1 k=1
{C} 3
{C} 1 k=1
{C} 2
[P] 2
[P] 3
{C} 3
```

**Comment by vchuravy** [comment]
`println` is a yield point. If you don't want to yield there you have to schedule a task to do the printing.
(Please direct questions like this to discourse.julialang.org)

**Comment by JeffBezanson** [comment]
Yes, I believe this is the intended behavior. But I can see a case for changing this for unbuffered channels, or adding a new kind of channel for this, for when you want the classic producer-consumer-style coroutine control flow.

**Comment by carlobaldassi** [comment]
Maybe a documentation issue then? I couldn't find an explanation of this behavior in the [Task section of the manual](https://docs.julialang.org/en/latest/manual/control-flow/#man-tasks-1) and thought it was unintended.
It's also problematic for my use case because the issue came up when the consumer in my library calls a function which can be customized by the user for their types. If they insert a `print` statement in their function all sorts of funky things may happen.
